rllib/algorithms/algorithm_config.py

```diff
@@ -2288,6 +2288,20 @@ def get_rollout_fragment_length(self, worker_index: int = 0) -> int:
                 diff = self.train_batch_size - int(
                     rollout_fragment_length
                 ) * self.num_envs_per_worker * (self.num_rollout_workers or 1)
+
+                if diff % self.num_envs_per_worker != 0:
```

Comment:
If diff is not a multiple integer of num_envs_per_worker, the simple logic that follows this is not guaranteed to adjust the rollout_fragment_length per workers to exactly get train_batch_size. For example for a train_batch_size = 5000, num_workers=32, and num_envs_per_worker =3, rollout_fragment_length has to be either 52, or 53 depending on worker_index (diff is 8). But the following assignment will assign 53 to workers 1, and 2 and the train_batch_size will become `2*53*3 + 30 * 52 * 3 = 4998` which is 2 steps shorter than the train_batch_size. Therefore the returned batch size would become `4998*2=9996` which is almost twice the train_batch_size specified in the config.